File: core/src/main/java/org/keycloak/representations/idm/MappingsRepresentation.java
package org.keycloak.representations.idm;

import java.util.List;
import java.util.Map;

/**
 * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
 * @version $Revision: 1 $
 */
public class MappingsRepresentation {
    protected List<RoleRepresentation> realmMappings;
    protected Map<String, ClientMappingsRepresentation> clientMappings;

    public List<RoleRepresentation> getRealmMappings() {
        return realmMappings;
    }

    public void setRealmMappings(List<RoleRepresentation> realmMappings) {
        this.realmMappings = realmMappings;
    }

    public Map<String, ClientMappingsRepresentation> getClientMappings() {
        return clientMappings;
    }

    public void setClientMappings(Map<String, ClientMappingsRepresentation> clientMappings) {
        this.clientMappings = clientMappings;
    }
}

File: js/libs/keycloak-admin-client/src/defs/mappingsRepresentation.ts
/**
 * https://www.keycloak.org/docs-api/11.0/rest-api/index.html#_mappingsrepresentation
 */
import type RoleRepresentation from "./roleRepresentation.js";

export default interface MappingsRepresentation {
  clientMappings?: Record<string, any>;
  realmMappings?: RoleRepresentation[];
}
